platform/state/platform_state.proto

```diff
@@ -0,0 +1,261 @@
+     *
+     * A temporary workaround until dynamic address books are supported.
+     * If present, the previous address book SHOULD reflect the address
+     * book from the preceding round.
+     */
+    proto.NodeAddressBook previous_address_book = 2;
+
+    /**
+     * The round of this state.
+     *
+     * MUST represent the handling of all transactions reaching
+     * consensus in all previous rounds. <br>
+     * The first state (genesis state) SHALL have a round of 0, because
+     * the first round is defined as round 1.
+     */
+    uint64 round = 3;
+
+    /**
+     * The running event hash.
+     *
+     * Computed by the consensus event stream. <br>
+     * SHOULD be deleted once the consensus event stream is retired.
+     */
+    bytes legacy_running_event_hash = 4;
+
+    /**
+     * The consensus timestamp for this signed state.
+     *
+     * MUST represent the consensus of the first transaction in this round.
+     */
+    proto.Timestamp consensus_timestamp = 5;
+
+    /**
+     * The version of the application software.
+     *
+     * MUST accurately reflect the software version that created this state.
+     */
+    SoftwareVersion creation_software_version = 6;
+
+    /**
+     * The epoch hash of this state.
+     *
+     * MUST be updated whenever emergency recovery is performed.
+     */
+    bytes epoch_hash = 7;
+
+    /**
+     * The next epoch hash.
+     *
+     * Used to update the epoch hash at the next round boundary. <br>
+     * SHOULD NOT be part of the hash and MUST NOT be serialized.
+     */
+    bytes next_epoch_hash = 8;
+
+    /**
+     * The number of non-ancient rounds.
+     *
+     * MUST accurately reflect the count of rounds considered non-ancient.
+     */
+    uint32 rounds_non_ancient = 9;
+
+    /**
+     * A snapshot of the consensus state at the end of the round.
+     *
+     * MUST be used for restart/reconnect.
+     */
+    ConsensusSnapshot consensus_snapshot = 10;
+
+    /**
+     * The time when the freeze starts.
+     *
+     * If null, a freeze SHOULD NOT be scheduled.
+     */
+    proto.Timestamp freeze_time = 11;
+
+    /**
+     * The last time a freeze was performed.
+     *
+     * If null, there has never been a freeze.
+     */
+    proto.Timestamp last_frozen_time = 12;
+
+    /**
+     * Null if birth round migration has not yet happened.
+     *
+     * Otherwise, MUST represent the software version that first
+     * enabled birth round mode.
+     */
+    SoftwareVersion first_version_in_birth_round_mode = 13;
+
+    /**
+     * The last round before the birth round mode was enabled.
+     *
+     * MUST be -1 if birth round mode has not yet been enabled.
+     */
+    uint64 last_round_before_birth_round_mode = 14;
+
+    /**
+     * The lowest judge generation before birth round mode was enabled.
+     *
+     * MUST be -1 if birth round mode has not yet been enabled.
+     */
+    uint64 lowest_judge_generation_before_birth_round_mode = 15;
```

Comment:
Both above uint64 type, if the value can be -1 (from the comments above), this should be int64 type.

Reply:
True. I'll fix it